### DISCUSSION TITLE
sqlalchemy.sql.select usage between 1.3 and 1.4

### DISCUSSION THREAD

**Discussion by mzagrabe** [sqlalchemy/sqlalchemy#11084] [Usage Questions] [answered]
Greetings!

Thank you for SQLAlchemy!

I am finally upgrading a project from 1.3 to 1.4 and am hitting a little snag and would appreciate some information and/or guidance.

I have the following code:

```
# 1.3 
for current_record in session.execute(
    select([CurrentRecord])
).fetchall():
    row_dict = dict(current_record)
```

...which I've changed to:

```
# 1.4 
for current_record in session.execute(
    'SELECT * FROM current_records'
).fetchall():
    row_dict = current_record._mapping
```

My question is about the `select([CurrentRecord])` changing to the literal SQL (`SELECT * FROM current_records`). Is there a way in 1.4 to maintain the sqlalchemy.sql.select code? That is, I find the `select([CurrentRecord])` to "look better" than the raw SQL. I'd like to keep it if possible.

I've tried some different permutations and also looked through the docs, but I don't see why the behavior changed for the `select` function.

Alternatively is it possible to use `_mapping` in an ORM sense?

```
current_records = CurrentRecord.query.all()
for current_record in current_records:
    # is this possible somehow?
    row_dict = current_record._mapping
```

Thank you for any help!

**Comment by CaselIT** [comment, marked as answer]
Hi,

Select was unified for orm and core usage, so the logic since sqlalchemy 1.4+ is that selecting an orm model from a session will give you orm instances, while using a connection will give you tuple like rows

So in 1.4+ to get back a tuple-like rows you would use a connection:
```py
for current_record in session.connection().execute(
    select(CurrentRecord)
).mappings().all():
    row_dict = current_record
```

above the use of `mappings()` is just to return dict like object instead of tuple like ones. It's eqivalet do doing `row._mapping` on each row

**Reply by mzagrabe** [reply]
Thank you for the prompt reply @CaselIT , and for the code example. Works great!
